broadinstitute/wfl
# [GH-877] New cli.py for new wfl-instance Terraform module

### Purpose


- https://broadinstitute.atlassian.net/browse/GH-877
- Supersedes #87 

### Changes


- New usage:
```
usage: cli.py [-h] [-d] COMMAND ENV INSTANCE [...]

deploy or connect to WFL infrastructure

positional arguments:
  COMMAND               one of [info, connect, deploy]
  ENV                   specify 'gotc-deploy/deploy/{ENV}' with one of [gotc-
                        dev, gotc-prod, aou]
  INSTANCE              specify the ID of the instance to use in the
                        environment

optional arguments:
  -h, --help            show this help message and exit
  -d, --dry-run         exit before COMMAND makes any remote changes
  -v VERSION, --version VERSION
                        specify the version to use instead of the 'version'
                        file
  --project PROJECT     specify the GCP project name instead of 'broad-{ENV}'
  --cloud-sql-name CLOUD_SQL_NAME
                        specify the Cloud SQL name instead of inferring from
                        labels
  --cluster-zone CLUSTER_ZONE
                        specify the zone for the GKE cluster instead of
                        inferring from CLUSTER_NAME
  --cluster-name CLUSTER_NAME
                        specify the GKE cluster name instead of inferring from
                        labels
  --cluster-namespace CLUSTER_NAMESPACE
                        specify the K8s namespace instead of '{INSTANCE}-wfl'
```
- Rendering can be directly accessed through its own script:
```
usage: render_ctmpl.py [-h] [-v VERSION] file

positional arguments:
  file                  CTMPL file to be rendered

optional arguments:
  -h, --help            show this help message and exit
  -v VERSION, --version VERSION
                        A specific version to use other than the root's
                        `version` file
```
- Made a util folder with misc.py to hold common logic between both scripts
- Used functions instead of classes

### Review Instructions


- On my branch you can:
  - Run the info command to see it infer arguments
  - Run the connect command (`connect gotc-dev dev`) to be connected to the brand new cloud sql instance in gotc-dev
  - Run deploy dry run (`deploy gotc-dev dev --dry-run`) to see it do everything that doesn't have remote side effects
